src/coreclr/jit/llvm.cpp

```diff
@@ -1896,7 +1802,252 @@ void Llvm::ConvertShadowStackLocalNode(GenTreeLclVarCommon* node)
+    }
+
+    if (callNode->gtCallArgs != nullptr)
+    {
+        for (GenTree* operand : callNode->Operands())
+        {
+            if (operand->IsArgPlaceHolderNode() || !operand->IsValue())
+            {
+                // Either of these situations may happen with calls.
+                continue;
+            }
+            fgArgTabEntry* curArgTabEntry = _compiler->gtArgEntryByNode(callNode, operand);
+            regNumber      argReg         = curArgTabEntry->GetRegNum();
+            if (argReg == REG_STK || curArgTabEntry->argType == TYP_BYREF) // TODO-LLVM: out and ref args
+            {
+                failFunctionCompilation();
+            }
+        }
+    }
+}
+
+GenTree* Llvm::createStoreNode(var_types nodeType, GenTree* addr, GenTree* nodeToStore, ClassLayout* structClassLayout)
+{
+    GenTree* storeNode;
+    if (nodeType == TYP_STRUCT)
+    {
+        storeNode = new (_compiler, GT_STORE_OBJ)
+            GenTreeObj(nodeType, addr, nodeToStore, structClassLayout);
+    }
+    else
+    {
+        storeNode = new (_compiler, GT_STOREIND) GenTreeStoreInd(nodeType, addr, nodeToStore);
+    }
+    return storeNode;
+}
+
+//------------------------------------------------------------------------
+// lowerCallToShadowStack: Lower the call, rewriting its arguments.
+//
+// This method has two primary objectives:
+//  1) Transfer the information about the arguments from arg info to explicit
+//     PutArgType nodes, to make it easy for codegen to consume it. Also, get
+//     rid of the late/non-late argument distinction, by sorting the inserted nodes
+//     in the original evaluation order, matching that of them in the signature.
+//  2) Rewrite arguments and the return to be stored on the shadow stack. We take
+//     the arguments which need to be on the shadow stack, remove them from the call
+//     arguments list, store their values on the shadow stack, at offsets calculated
+//     in a simple increasing order, matching the signature. We also rewrite returns
+//     that must be on the shadow stack, see "lowerCallReturn".
+//
+void Llvm::lowerCallToShadowStack(GenTreeCall* callNode, CORINFO_SIG_INFO& calleeSigInfo)
+{
+    // rewrite the args, adding shadow stack, and moving gc tracked args to the shadow stack
+    unsigned shadowStackUseOffest = 0;
+
+    fgArgInfo*                 argInfo    = callNode->fgArgInfo;
+    unsigned int               argCount   = argInfo->ArgCount();
+    fgArgTabEntry**            argTable   = argInfo->ArgTable();
+    std::vector<OperandArgNum> sortedArgs = std::vector<OperandArgNum>(argCount);
+    OperandArgNum*             sortedData = sortedArgs.data();
+
+    GenTreeCall::Use* lastArg;
+    GenTreeCall::Use* insertReturnAfter;
+    GenTreeCall::Use* callThisArg = callNode->gtCallThisArg;
+
+    callNode->ResetArgInfo();
+    callNode->gtCallThisArg = nullptr;
+    argCount                = argInfo->ArgCount();
+    argTable                = argInfo->ArgTable();
```

Comment:
This appears redundant with the initialization done at declaration.

Reply:
Thanks